File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        // Since we need to return the indices we can't sort the array
        unordered_map<int, int> mpp;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            int num = nums[i];
            int moreNeeded = target - num;
            if (mpp.find(moreNeeded) != mpp.end()) {
                return {mpp[moreNeeded], i};
            }
            mpp[num] = i;
        }
        return {-1, -1};
    }
};

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */

var twoSum = function (nums, target) {
    let mpp = new Map();
    let n = nums.length;
    for (let i = 0; i < n; i++) {
        let num = nums[i];
        let moreNeeded = target - num;
        if (mpp.has(moreNeeded)) {
            return [mpp.get(moreNeeded), i];
        }
        mpp.set(num, i);
    }
    return [-1, -1];
};